### PR TITLE
Really treating warnings as errors

### PR DESCRIPTION
**Pull request by moozzyk**

### CODE REVIEW

**Comment by victorhurdugaci** [comment]
:shipit:

File: build/Dnx.Native.Settings [hunk 1]
```diff
@@ -24,7 +24,7 @@
       <PrecompiledHeader>Use</PrecompiledHeader>
       <WarningLevel>Level4</WarningLevel>
       <PreprocessorDefinitions>_CONSOLE;%(PreprocessorDefinitions);$(Defines)</PreprocessorDefinitions>
-      <TreatWarningAsError Condition="'$(TreatWarningsAsErrors)' != ''">true</TreatWarningAsError>
+      <TreatWarningAsError Condition="'$(TreatWarningsAsErrors)' == ''">true</TreatWarningAsError>
```
Comment (Eilon) [inline build/Dnx.Native.Settings]: Kind of confused... isn't this now backwards?

1. Why would we enable TWAE only if the setting is *empty*?
2. Why not just force it to always be on? (That is, don't have a setting for it.)

Comment (moozzyk) [inline build/Dnx.Native.Settings]: It is empty by default. This allows you to override if you need for any reason (e.g. when hacking on it on your own box)

Comment (Eilon) [inline build/Dnx.Native.Settings]: Makes sense, thanks!

**Pull request merged**